src/magicgui/type_map/_type_map.py

```diff
@@ -252,15 +252,15 @@ def _pick_widget_type(
 
     # look for subclasses
     for registered_type in _TYPE_DEFS:
-        if _type == registered_type or safe_issubclass(_type, registered_type):
-            _cls, opts = _TYPE_DEFS[registered_type]
-            return _cls, {**options, **opts}
+        if type_ == registered_type or safe_issubclass(type_, registered_type):
+            cls_, opts = _TYPE_DEFS[registered_type]
+            return cls_, {**options, **opts}
 
     if is_result:
-        _widget_type = match_return_type(_type)
-        if _widget_type:
-            _cls, opts = _widget_type
-            return _cls, {**options, **opts}
+        widget_type_ = match_return_type(type_)
+        if widget_type_:
+            cls_, opts = widget_type_
+            return cls_, {**opts, **options}
```

Comment:
and here same change, but it is not the code branch reached by the original issue.